src/components/Modal/index.tsx

```diff
@@ -0,0 +1,99 @@
+import Icon from '@components/Icon'
+import styled from '@emotion/styled'
+import { css } from '@emotion/react'
+import React, { useState } from 'react'
+import { COLORS } from '@utils/constants/colors'
+
+interface IModal {
+  title?: string
+  confirm?: boolean
+}
+
+interface IProps {
+  open: boolean
+}
+
+const Modal: React.FC<IModal> = ({ children, title = '', confirm = false }) => {
+  if (React.Children.count(children) < 2) {
+    return null
+  }
+
+  const [open, setOpen] = useState(false)
+
+  const handleClickOpen = () => {
+    setOpen(true)
+  }
+
+  const handleClose = () => {
+    setOpen(false)
+  }
+
+  return (
+    <>
+      <div onClick={handleClickOpen}>
+        {React.Children.count(children) > 1 &&
+          React.Children.toArray(children)[0]}
+      </div>
+      <StyledModal open={open}>
+        <ModalHeader>
+          {title} {/*Text 컴포넌트 적용 예정 */}
+          <Icon
+            name={'close'}
+            style={{ position: 'absolute', top: '16px', right: '16px' }}
+            onIconClick={handleClose}
+          />
+        </ModalHeader>
+        <ModalContent>
+          {React.Children.count(children) > 1 &&
+            React.Children.toArray(children)[1]}
+          {confirm && (
+            <ModalBottom onClick={handleClose}>
+              {React.Children.count(children) === 3 &&
+                React.Children.toArray(children)[2]}
+            </ModalBottom>
+          )}
+        </ModalContent>
+      </StyledModal>
+    </>
+  )
+}
+
+const StyledModal = styled.div`
+  ${({ open }: IProps) => {
+    return open
+      ? css`
+          display: block;
+        `
+      : css`
+          display: none;
+        `
+  }}
+  position: fixed;
+  top: 0;
+  width: 425px;
```

Comment:
거의 모든 영역을 모달이 덮게 될텐데 
100%으로만 주면 최상위 요소가 max, min width가 적용되어 있으니 알아서 이쁘게 줄어들 거 같습니다.
```suggestion
  width: 100%;
```

Reply:
100%로 했을 때 부모 container의 width보다 커지는 이슈가 있습니다.
fixed 때문인 것으로 추정되는데 고민해보겠습니다..!!